plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/preferences/OCBinary.java

```diff
@@ -35,20 +48,30 @@ public String getName() {
 		return extensions;
 	};
 
-	public String getLocation() {
+	/**
+	 * Tries to find the binary on the system path.
+	 * 
+	 * @return the location as found on the system path.
+	 */
+	public String findLocation() {
 		if( locationBinary == null ) {
 			locationBinary = new CommandLocationBinary("oc");
-			locationBinary.addPlatformLocation(Platform.OS_LINUX, "/usr/bin/oc");
+			locationBinary.addPlatformLocation(Platform.OS_LINUX, OC_DEFAULTLOCATION_LINUX);
 			locationBinary.setDefaultPlatform(Platform.OS_LINUX);
 		}
 		return locationBinary.findLocation();
 	}
-	
-	public static OCBinaryName getInstance() {
-		if (SystemUtils.IS_OS_WINDOWS) {
-			return WINDOWS;
-		} else {
-			return OTHER;
+
+	/**
+	 * Returns the location from preferences or looks it up on the path.
+	 * 
+	 * @return
+	 */
+	public String getLocation() {
+		String location = OpenShiftCorePreferences.INSTANCE.getOCBinaryLocation();
+		if (StringUtils.isBlank(location)) {
+			location = findLocation();
 		}
+		return location;
```

Comment:
Oh, yes. I would vote for getDefaultLocation(), name findLocation() remained obscure to me. But it is ok, as is done.